Data-Structures/CircularQueue.ts

```diff
@@ -2,7 +2,13 @@
  * @class CircularQueue 
  * @description Circular implementation of a queue 
 */
-export class CircularQueue<T> {
+export interface Queue<T> {
```

Comment:
No. That way you still get rid of the non-circular Queue implementation. I want there to be two Queue classes, one circular and the other one not circular. They may both implement a common interface.